### ISSUE TITLE
Hidden columns become visible when we run updateSettings

### ISSUE THREAD

**Issue by rayokao** [handsontable/handsontable#4121]
### Description
Hidden columns become visible when cell is edited. This occurs only when sorting has been enabled and seems to have something to do with the Observer-plugin. This problem will not occur if the handsontable instance has been intialized with data (via settings.data or loadData) prior to editing. This problem seems not to be present in handsontable version 1.3.4.

### Steps to reproduce
1. Configure hiddenColumns and enable columnSorting. Do not provide data for the constructor.
2. Edit some cell and select another cell.
3. Hidden columns become visible.

The problem occurs when the grid data has not been initialized:
![hiddencolumns sorting failure](https://cloud.githubusercontent.com/assets/5380050/23747699/4f34c366-04c9-11e7-9b78-50c8b2515b59.gif)

The problem will not occur if the data is loaded prior to editing:
![hiddencolumns sorting success2](https://cloud.githubusercontent.com/assets/5380050/23748299/ab31baa0-04cb-11e7-8aa4-f085cfbcc99f.gif)

### Demo
https://jsfiddle.net/rayokao/qpxbsy2m/

### Your environment
* Handsontable version: 1.10.1 (pro)
* Browser Name and version: Chrome 56
* Operating System: Windows 10

**Comment by rayokao** [comment]
Initializing the grid with an empty array ( settings.data : []) instead of null seems to fix the issue.

**Comment by AMBudnik** [comment]
It is but I think that still we can add the exception inside the source code to avoid this kind of issues.

**Comment by AMBudnik** [comment]
Hi @rayokao 
Can you check if this issue is still replicable? The demo does not seem to be the same as the one in the recording.

**Comment by rayokao** [comment]
Updated the demo-link to the correct jsFiddle. The same problem seems to be present in the newest versions as well. Not sure if this is a valid bug though. (Note that the jsFiddle works until 1.11.0 after that Dom.addEvent needs to be dom.addEvent)

**Comment by AMBudnik** [comment]
Works the same in v 6.1.1

**Comment by aninde** [comment]
This bug is still replicable at current version 7.1.1.

**Comment by aninde** [comment]
This problem will be fixed with the new index mapper.  It will be released soon.
We are currently testing a version with many breaking changes, I recommend observing information from issue [#5751.](https://github.com/handsontable/handsontable/issues/5751)

**Demo**
version 8.0.0rc https://jsfiddle.net/aninde/oc34jmnq/

**Environment**
Handsontable version:8.0.0rc
Browser Name and version: Google Chrome version 76.0.3809.100
Operating System: macOS Mojave 10.14.6

**Comment by AMBudnik** [comment]
Hi @rayokao 

now you can try our official 8.0.0-beta1 where the issue is solved https://jsfiddle.net/5h1qy20n/

```html
</style><script src="https://cdn.jsdelivr.net/gh/handsontable/handsontable@release/8.0.0-beta.1/dist/handsontable.full.js"></script><link type="text/css" rel="stylesheet" href="https://cdn.jsdelivr.net/gh/handsontable/handsontable@release/8.0.0-beta.1/dist/handsontable.full.css">
```
`npm i handsontable@beta`

**Comment by AMBudnik** [comment]
There's no issue with 8.0.0-beta2-rev1 https://jsfiddle.net/xbmr9y86/  
When I edit a cell hiddeColumns stay hidden (the header may trick you cause of #6851) but the amount of columns do not change.

**Issue solved.**

**Comment by jeremy-smith-maco** [comment]
Hey. I am having this issue too as the title describes but the reproduction steps are different. If you want me to make a separate issue then let me know. This issue seems to be happening in the 8.0.0-beta2-rev1 as well.

My reproduction steps:
1. Enable HiddenColumns
2. Manually hide some columns using the plugin methods.
3. Call updateSettings()
4. Columns reappear

They do not reappear if specified in the original options for the table.

Example: http://jsfiddle.net/meja7421/ (You will see the columns reappear after 3 seconds when updateSettings() is run)

Thanks

**Comment by AMBudnik** [comment]
Hi @jeremy-smith-maco 

that scenario is already mentioned here https://github.com/handsontable/handsontable/issues/5571#issuecomment-625711788 but I really appreciate your feedback. I will update you once we solve it. I've tested it with the newest revision no 10 and it is still working the same way http://jsfiddle.net/AMBudnik/4wzjfuks/

**Comment by jeremy-smith-maco** [comment]
Ah, okay. I didn't check that issue. Thanks @AMBudnik

**Comment by pnowak** [comment]
The original issue is fixed - https://jsfiddle.net/4dnkmwj1/

But issue from @jeremy-smith-maco  is still present - https://jsfiddle.net/nacdj61v/

**Comment by AMBudnik** [comment]
@pnowak is right, @jeremy-smith-maco I moved your use case to a separate issue https://github.com/handsontable/handsontable/issues/7165

**Comment by AMBudnik** [comment]
Thank you for the demo @pnowak 

@rayokao we can close this issue as fixed in v 8.0.0. Thank you for your report and your patience.
